### PR TITLE
build: provide makefile for build process

### PR DESCRIPTION
**Pull request by bromiesTM**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	-x "apps-*/*/node_modules**" \
+	-x "apps-*/*/tests**" \
+	-x "**/cypress/**" \
+	-x "*.git*" \
+	-x "*.editorconfig*" \
+	-x "themes/nc-ionos-theme/README.md" \
+	-x "themes/nc-ionos-theme/IONOS**")
+
+build: simplesettings_app user_oidc_app ionos_theme add_config_partials zip_dependencies ## All of the above
+	@echo "Everything done"
```
Comment (thlehmann-ionos) [inline Makefile:89]: Let's print the state more verbose: `Everything done for a release`

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+build_locally: simplesettings_app user_oidc_app ionos_theme zip_dependencies ## All of the above without copying config files
```
Comment (thlehmann-ionos) [inline Makefile:91]: Let's describe the help better: `Build all apps/themes for local development`

Comment (thlehmann-ionos) [inline Makefile:91]: Zip is not needed for local development.

**Review by thlehmann-ionos — CHANGES_REQUESTED** [review]
See the comments. Most of them resolved in [`tl/review/kh/dev/build-nc-as-makefile`](https://github.com/IONOS-Productivity/nc-config/commits/tl/review/kh/dev/build-nc-as-makefile)

File: Makefile (hunk 2 above)
Comment (bromiesTM) [reply Makefile]: :heavy_check_mark:  Done with 22cfd86

Comment (bromiesTM) [reply Makefile]: :heavy_check_mark: Done with 8062549

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	-x "apps-*/*/src**" \
+	-x "apps-*/*/node_modules**" \
+	-x "apps-*/*/tests**" \
+	-x "**/cypress/**" \
+	-x "*.git*" \
+	-x "*.editorconfig*" \
+	-x "themes/nc-ionos-theme/README.md" \
+	-x "themes/nc-ionos-theme/IONOS**")
+
+build: simplesettings_app user_oidc_app ionos_theme add_config_partials zip_dependencies ## All of the above
```
[the remark this answers is not in the document]
Comment (bromiesTM) [reply Makefile]: :heavy_check_mark: Done with 
https://github.com/IONOS-Productivity/nc-config/compare/d949579fbdbef395862f88c7150b6fbb2c8ab8d0..22cfd86aca2f41adf459175fd9a683b54ff38b88

File: Makefile (hunk 1 above)
Comment (bromiesTM) [reply Makefile]: :heavy_check_mark: with 
https://github.com/IONOS-Productivity/nc-config/compare/d949579fbdbef395862f88c7150b6fbb2c8ab8d0..22cfd86aca2f41adf459175fd9a683b54ff38b88

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,92 @@
+# SPDX-FileCopyrightText: 2024 Kai Henseler <kai.henseler@strato.de>
```
[the remark this answers is not in the document]
Comment (bromiesTM) [reply Makefile:1]: :heavy_check_mark: with
https://github.com/IONOS-Productivity/nc-config/compare/d949579fbdbef395862f88c7150b6fbb2c8ab8d0..22cfd86aca2f41adf459175fd9a683b54ff38b88

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+	@echo "Usage: make [target]"
+	@echo ""
+	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z_-]+:.*?## / {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
+# HELP
+# This will output the help for each task
+# thanks to https://marmelab.com/blog/2016/02/29/auto-documented-makefile.html
+.DEFAULT_GOAL := help
+
+simplesettings_app: ## Install and build simplesettings app
+	@(cd ../apps-custom/simplesettings && \
```
[the remark this answers is not in the document]
Comment (bromiesTM) [reply Makefile]: :heavy_check_mark: with
https://github.com/IONOS-Productivity/nc-config/compare/d949579fbdbef395862f88c7150b6fbb2c8ab8d0..22cfd86aca2f41adf459175fd9a683b54ff38b88

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,92 @@
+# SPDX-FileCopyrightText: 2024 Kai Henseler <kai.henseler@strato.de>
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+TARGET_PACKAGE_NAME=easy-storage.zip
+
+.PHONY: all
```
[the remark this answers is not in the document]
Comment (bromiesTM) [reply Makefile]: :heavy_check_mark: with
https://github.com/IONOS-Productivity/nc-config/compare/d949579fbdbef395862f88c7150b6fbb2c8ab8d0..22cfd86aca2f41adf459175fd9a683b54ff38b88

**Comment by thlehmann-ionos** [comment]
Rebased onto master

**Review by thlehmann-ionos — APPROVED** [review]

**Pull request merged**